src/utils.rs

```diff
@@ -52,6 +52,25 @@ pub fn do_write_line(
     }
 }
 
+/// Retrieves the modification time of the input file specified by file path
+/// If an error occurs, it returns the current system time
+pub fn get_modification_time(file_path: &str) -> String {
```

Comment:
Perhaps you can make the `file_path` argument a `std::path::Path`?
I *think* this would avoid having to call `to_string_lossy()` on the `from` and `to` parameters.

Reply:
Although this will avoid the need for calling `to_string_lossy()`, we will now have to convert `&str` to `std::path::Path` and `&OsString` to `std::path::Path`.

Reply:
`std::path::Path::new()` takes a string slice and constructs a `Path`. But this wouldn't add much value, unless `Params.from` and `Params.to` were already of type `Path` (which would be an interesting improvement, but outside of the scope of this PR).